src/meltano/core/plugin/singer/catalog.py

```diff
@@ -87,38 +82,34 @@ def match(self, tap_stream_id: str, breadcrumb: list[str] | None = None) -> bool
         return result
 
 
-class MetadataRule(CatalogRule):  # noqa: D101
-    def __init__(
-        self,
-        tap_stream_id: str | list[str],
-        breadcrumb: list[str] | None,
-        key: str,
-        *,
-        value: bool,
-        negated: bool = False,
-    ):
-        """Create a metadata rule for a stream and property."""
-        super().__init__(tap_stream_id, breadcrumb, negated=negated)
-        self.key = key
-        self.value = value
+@dataclasses.dataclass
```

Comment:
Dataclasses are slightly faster:

```python
import timeit

code = """
rules = [
    MetadataRule(
        "my_stream",
        [],
        "selected",
        value=True,
    ),
    MetadataRule(
        "my_stream",
        ["properties", "prop", "properties", "*"],
        "selected",
        value=True,
    ),
    MetadataRule(
        "my_stream",
        ["properties", "prop"],
        "selected",
        value=True,
    ),
]
"""

setup = """
from meltano.core.plugin.singer.catalog import MetadataRule
"""

print(timeit.timeit(stmt=code, setup=setup, number=1000000))

# 0.6132017500021902 (normal class)
# 0.3916304169979412 (dataclass)
```

Reply:
Using dataclasses makes the intent of the code more clear too, and less error prone